types/detect/detect.go

```diff
@@ -22,8 +22,9 @@ type Signature interface {
 }
 
 type SignatureContext struct {
-	Callback SignatureHandler
-	Logger   Logger
+	Callback      SignatureHandler
+	Logger        Logger
+	GetDataSource func(namespace string, id string) (DataSource, bool)
```

Comment:
Maybe create a struct for the namespace and ID.
It would reduce the amount of functions needed to implement in the interface and make it easier to change names and such in the future.

Reply:
How would it reduce the number of functions?
However I can see the merits if this function changes using a struct will reduce the chance for breaking changes on the calling side.
On the implementing side this is an interface we implement in the engine's side, the signature author only consumes the interface as a user not as an implementor.